src/main/java/com/gluonhq/substrate/target/LinuxTargetConfiguration.java

```diff
@@ -63,6 +63,13 @@ public class LinuxTargetConfiguration extends PosixTargetConfiguration {
             "java", "nio", "zip", "net", "prefs", "j2pkcs11", "sunec", "extnet", "fdlibm",
             "fontmanager", "javajpeg", "lcms", "awt_headless", "awt"
     );
+    /**
+     * fdlibm no longer required, unsure since when
```

Comment:
`fdlibm` is also no longer required on MacOSTarget.
Therefore, we should probably keep one single `staticJavaLibs` per target with it, and then in `getTargetSpecificJavaLinkLibraries()` add another filter for it, something like:
```
private List<String> getTargetSpecificJavaLinkLibraries() {
        return Stream.concat(getStaticJavaLibs().stream(), getOtherStaticLibs().stream())
                .filter(lib -> {
                    if (projectConfiguration.usesJDK11()) {
                        return true;
                    } else if (javaVersion.getMajor() < 21) {
                        return !lib.contains("sunec");
                    }
                    return !lib.contains("sunec") && !lib.contains("fdlibm");
                })
                .map(this::getLinkLibraryOption)
                .collect(Collectors.toList());
    }
```

Reply:
I'm not sure it's that straightforward.
For Window, both `mswsock` in the "other libs" and `extnet` in the "java libs" are new in the list for GraalVM for JDK 21.
I guess it's possible in your setup if for every future addition, there is a matching filter for every version below that, but I think that'll quickly get quite ugly and requires 2 separate modifications to add a library.

An alternative would be to build the list additively in the `getStaticJavaLibs` and `getOtherStaticLibs` methods e.g.
```java
    List<String> getStaticJavaLibs() {
        int major = projectConfiguration.getJavaVersion().getMajor();
        List<String> libs = new ArrayList<>(List.of(
                "j2pkcs11", "java", "net", "nio", "prefs", "zip", "sunmscapi"));
        if (major == 11) {
            libs.add("sunec");
        }
        if (major < 21) {
            libs.add("fdlibm");
        }
        if (major >= 21) {
            libs.add("extnet");
        }
        return libs;
    }
```

But that does away with the static list that's currently used.

Reply:
Right, I missed the `extnet` addition.

We want to have the static libs declared on top, but I like the change you propose, so what about (for Windows, that is):

```
    private static final List<String> staticJavaLibsBase = Arrays.asList(
            "j2pkcs11", "java", "net", "nio", "prefs", "zip", "sunmscapi");
    private static final List<String> staticJavaLibs11 = List.of("sunec");
    private static final List<String> staticJavaLibs20 = List.of("fdlibm");
    private static final List<String> staticJavaLibs21 = List.of("extnet");


@Override
    List<String> getStaticJavaLibs() {
        int major = projectConfiguration.getJavaVersion().getMajor();
        List<String> libs = staticJavaLibsBase;
        if (major == 11) {
            libs.addAll(staticJavaLibs11);
        }
        if (major < 21) {
            libs.addAll(staticJavaLibs20);
        }
        if (major >= 21) {
            libs.addAll(staticJavaLibs21);
        }
        return libs;
    }
```

If you do it like this in the other targets, we could get rid of the filter in `getTargetSpecificJavaLinkLibraries` to remove `sunec`.
Also, don'f forget about `fdlibm` in macOS.

Reply:
I've appliled this in the `AbstractTargetConfiguration` class as well, assuming it's necessary for the targets outside of Linux/Windows/MacOS as well.
It's applied like this for now, one thing I don't like that much is the amount of duplication that's currently in there.

It would be nice having just a single list of libraries wrapped in a version range..
```java
List<JavaLib> staticJavaLibs = List.of(
        // applied to all versions
        JavaLib.of("java"),
        // applied up to specific version
        JavaLib.upTo(11, "sunec"),
        // applied from specific version
        JavaLib.from(21, "extnet"),
        // applied within version range, not currently applicable
        JavaLib.range(11, 21, "example")
    );
```
That's a single field with everything in it, after which it can easilly be filtered in the abstract parent as such:
```java
int major = projectConfiguration.getJavaVersion().getMajor();
staticJavaLibs.stream()
        .filter(l -> l.inRange(major))
        .map(JavaLib::getLibName)
        .toList();
```
I feel this is getting a bit off track though.

Reply:
yes, and on top of that, we are missing one important thing (sorry, my bad): the order of the libraries does matter. 

So we can't just remove them (fdlibm, sunec) from the initial ordered list, and add them at the end when they are needed, that won't work. 

That's why we had the filter to remove sunec in the first place, and what my initial proposal was about.

So I'd say we should go back to it:
```
private List<String> getTargetSpecificJavaLinkLibraries() {
       int major = projectConfiguration.getJavaVersion().getMajor();
       return Stream.concat(getStaticJavaLibs().stream(), getOtherStaticLibs().stream())
                .filter(lib -> {
                    if (major == 11) {
                        return true; // all are needed
                    } else if (major < 21) {
                        return !lib.contains("sunec"); // remove sunec, keep fdlibm
                    }
                    return !lib.contains("sunec") && !lib.contains("fdlibm"); // remove sunec, fdlibm
                })
                .map(this::getLinkLibraryOption)
                .collect(Collectors.toList());
    }
```

For Windows, you can simply add extnet (assuming this can go at the end of the list):
```
@Override
    List<String> getStaticJavaLibs() {
        List<String> libs = new ArrayList<>(staticJavaLibs);
        if (projectConfiguration.getJavaVersion().getMajor() >= 21) {
            libs.addAll(staticJavaLibs21);
        }return libs;
    }
```
Then linux and Mac don't need changes on this matter.

Sorry for the confusion, but does this make sense to you?

Reply:
What about the single list with version bounds? That would preserve order and IMO is more readable and maintainable than the filter.

Reply:
yes, that could work as well, of course, seems more complex to implement, but you are right about future maintenance.